### PR TITLE
Optimise MapBlockMesh functions

### PR DESCRIPTION
**Pull request by gregorycu**
Feeback in GitHub only please
Optimise the following functions:
MapBlockMesh::MapBlockMesh
MapBlockMesh::getTileInfo
MapBlockMesh::makeFastFace
MapBlockMesh::getSmoothLightCombined
Speedup is about 33% for these functions.

### CODE REVIEW

File: src/voxel.h [hunk 1]
```diff
@@ -569,6 +580,8 @@ class VoxelManipulator /*: public NodeContainer*/
 	*/
 	u8 *m_flags;
 
+	static const MapNode ContentIgnoreNode;
```
Comment (SmallJoker) [inline src/voxel.h]: Using `MapNode(CONTENT_IGNORE)` would result in clearer codes IMO

Comment (gregorycu) [inline src/voxel.h]: But it results in slower code (means I can't return by ref).
What settings did you mean?

Comment (SmallJoker) [inline src/voxel.h]: * Functions

Couldn't you optimize one of the already existing functions instead of adding a new?

Comment (gregorycu) [inline src/voxel.h]: It's possible, but the function doesn't check for out-of-bounds indexing, which means I'd be making the functions less safe.

All it takes is one person who "thinks" they know what the functions do to rely on their old behaviour, and we have a crash (and if we're unlucky, an uncommon one).

File: src/voxel.h [hunk 2]
```diff
@@ -413,10 +413,21 @@ class VoxelManipulator /*: public NodeContainer*/
 	}
 	// Stuff explodes if non-emerged area is touched with this.
 	// Emerge first, and check VOXELFLAG_NO_DATA if appropriate.
-	MapNode & getNodeRefUnsafe(v3s16 p)
+	MapNode & getNodeRefUnsafe(const v3s16 &p)
 	{
 		return m_data[m_area.index(p)];
 	}
+
+	const MapNode & getNodeRefUnsafeCheckFlags(const v3s16 &p)
```
Comment (Zeno-) [inline src/voxel.h]: Why is this added instead of using getNodeNoEx() ?

**Comment by Zeno-** [comment]
@gregorycu can you answer my question please so I can merge this?

**Comment by Zeno-** [comment]
577701cabd6382bb97dc05e9c9244b5ce8202ca3
